src/node_i18n.cc

```diff
@@ -62,6 +72,296 @@ bool flag_icu_data_dir = false;
 
 namespace i18n {
 
+Local<Value> ICUException(Environment* env,
+                          UErrorCode status,
+                          const char* msg) {
+  Local<String> estring = OneByteString(env->isolate(), u_errorName(status));
+  if (msg == nullptr || msg[0] == '\0')
+    msg = "Unspecified ICU Exception";
+  Local<String> cons =
+      String::Concat(estring, FIXED_ONE_BYTE_STRING(env->isolate(), ", "));
+  cons = String::Concat(cons, OneByteString(env->isolate(), msg));
+
+  Local<Value> e = Exception::Error(cons);
+  Local<Object> obj = e->ToObject(env->isolate());
+  obj->Set(env->errno_string(), Integer::New(env->isolate(), status));
+  obj->Set(env->code_string(), estring);
+  return e;
+}
+
+void ThrowICUError(Environment* env, UErrorCode status, const char* msg) {
+  HandleScope scope(env->isolate());
+  env->isolate()->ThrowException(ICUException(env, status, msg));
+}
+
+// TODO(jasnell): This could potentially become a member of MaybeStackBuffer
+// at some point in the future. Care would need to be taken with the
+// MaybeStackBuffer<UChar> variant below.
+MaybeLocal<Object> AsBuffer(Isolate* isolate,
+                            MaybeStackBuffer<char>* buf,
+                            size_t len) {
+  if (buf->IsAllocated()) {
+    MaybeLocal<Object> ret = Buffer::New(isolate, buf->out(), len);
+    buf->Release();
```

Comment:
I think this should be `if (!ret.Empty()) buf->Release();` - it's leaking memory now when the buffer can't be created.